### DISCUSSION TITLE
Memory Access Violation - Python Net 3.0.2 - Multi-threaded Dot Net application

### DISCUSSION THREAD

**Discussion by crazyoutlook** [pythonnet/pythonnet#2444] [Q&A]
### Environment

-   Pythonnet version: 3.0.2
-   Python version: 3.10
-   Operating System: Windows Server 2019 Datacenter
-   .NET Runtime: 6.0

### Details

-   Describe what you were trying to get done.

Running multi-threaded process in Dot net using Pythonet. I have a dot net application which used Pythonnet to invoke some python methods. As long as I am running single threaded process of my application it works fine. When I run multi-threaded application then I am getting error in Pythonnet. 

Fatal error. System.AcccessViolationExcepriom: Attempted to read or write protected memory. This is often an indication that other memory is corrupt:
    at Python.runtime.Runtime.PyErr_Clear()
     at Python.runtime.Exception.PyErr_Clear()
      at Python.Runtime.PythonEngine,Initialize(System.Collection.Generic.IEnumerable 1<System.String>

-   What commands did you run to trigger this issue? If you can provide a
    [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
    this will help us understand the issue.

Running multi-threaded process in Dot net using Pythonet. 

-   If there was a crash, please include the traceback here.

Fatal error. System.AcccessViolationExcepriom: Attempted to read or write protected memory. This is often an indication that other memory is corrupt:
    at Python.runtime.Runtime.PyErr_Clear()
     at Python.runtime.Exception.PyErr_Clear()
      at Python.Runtime.PythonEngine,Initialize(System.Collection.Generic.IEnumerable 1<System.String>

**Comment by crazyoutlook** [comment]
Hello Team, please guide me with this issue, provide some insights on how to resolve this issue. Can we do multithreading? As of now I am using SINGETON class to create only 1 object of python net.  And I am using that object to make all AI Model calls.

**Comment by tabs** [comment]
同样遇到了这个问题，多线程能力弱鸡
